scheduler/plugin/plugin_test.go

```diff
@@ -1807,3 +1886,48 @@ func requireError(t *testing.T, err error, expectedError string) {
 		t.Fatalf("unexpected error: want %v, have %v", want, have)
 	}
 }
+
+type waitingPod struct {
```

Comment:
why do we have to implement `waitingPod` here as well in addition to scheduler/test/testdata.go?